### PR TITLE
Log cleanup

### PR DESCRIPTION
**Pull request by dwalintukan**

### CODE REVIEW

File: src/db/nedb.js [hunk 1]
```diff
@@ -63,6 +64,29 @@ class DBHelper {
...
+    if (!_.isEmpty(fields)) {
+      fieldsObj = {};
+      _.each(fields, (field) => {
+        fieldsObj[field] = 1;
+      });
+    }
+
+    const found = await db.findOne(query, fieldsObj);
+    if (!found) {
+      const filename = db.nedb.filename;
```
Comment (alex-cory) [inline src/db/nedb.js]: does destructuring not work?
`const { filename } = db.nedb;`

File: src/db/nedb.js [hunk 2]
```diff
@@ -63,6 +64,29 @@ class DBHelper {
...
+  * Returns the fields of the object in one of the tables searched by the query.
+  * @param db The DB table.
+  * @param query {Object} The query by items.
+  * @param fields {Array} The fields to return for the found item in an array.
+  */
+  static async findOne(db, query, fields) {
+    let fieldsObj;
+    if (!_.isEmpty(fields)) {
+      fieldsObj = {};
+      _.each(fields, (field) => {
```
Comment (alex-cory) [inline src/db/nedb.js]: we might have to have this on multiple lines so it's not returning each time, but maybe test out to see if we can do:
`_.each(fields, (field) => fieldsObj[field] = 1)`

also, can't we use reduce for this?
`const fieldsObj = (fields || []).reduce((obj, field) => obj[field] = 1, {});`

might have to be:
`const fieldsObj = (fields || []).reduce((obj, field) => { obj[field] = 1 }, {});`

File: src/sync/index.js [hunk 3]
```diff
@@ -75,6 +69,12 @@ function sequentialLoop(iterations, process, exit) {
   return loop;
 }
 
+const startSync = async () => {
```
[the remark this answers is not in the document]
Comment (dwalintukan) [reply src/sync/index.js]: you are correct. fixed

File: src/db/nedb.js (hunk 1 above)
Comment (dwalintukan) [reply src/db/nedb.js]: fixed

File: src/db/nedb.js (hunk 2 above)
Comment (dwalintukan) [reply src/db/nedb.js]: reduce does not work for this. shortened it without the brackets

**Pull request merged**